toolboxes/feel/feelmodels/multifluid/surfacetensionforcemodel.hpp

```diff
@@ -0,0 +1,101 @@
+{
+    typedef SurfaceTensionForceModel<LevelSetType, FluidMechanicsType> self_type;
+    typedef InterfaceForcesModel<LevelSetType, FluidMechanicsType> super_type;
+public:
+    typedef typename super_type::levelset_type levelset_type;
+    typedef typename super_type::levelset_ptrtype levelset_ptrtype;
+
+    typedef typename super_type::fluidmechanics_type fluidmechanics_type;
+    typedef typename super_type::fluidmechanics_ptrtype fluidmechanics_ptrtype;
+
+    typedef typename super_type::mesh_type mesh_type;
+
+    typedef typename levelset_type::space_vectorial_type space_type;
+    typedef std::shared_ptr<space_type> space_ptrtype;
+
+    typedef typename space_type::element_type element_type;
+    typedef typename space_type::element_ptrtype element_ptrtype;
+
+    static const uint16_type Dim = levelset_type::nDim;
+
+    //--------------------------------------------------------------------//
+    // Construction
+    SurfaceTensionForceModel() = default;
+    SurfaceTensionForceModel( SurfaceTensionForceModel const& i ) = default;
+
+    void build( std::string const& prefix, levelset_ptrtype const& ls, fluidmechanics_ptrtype const& fm = fluidmechanics_ptrtype() ) override;
+
+    std::shared_ptr<std::ostringstream> getInfo() const override;
+
+    void loadParametersFromOptionsVm();
+    //--------------------------------------------------------------------//
+    double surfaceTensionCoeff() const { return M_surfaceTensionCoeff; }
+    void setSurfaceTensionCoeff( double k ) { M_surfaceTensionCoeff = k; }
+
+    //--------------------------------------------------------------------//
+    //--------------------------------------------------------------------//
+    //--------------------------------------------------------------------//
+private:
+    void updateInterfaceForcesImpl( element_ptrtype & F ) const override;
+
+    //--------------------------------------------------------------------//
+    double M_surfaceTensionCoeff;
+
+};
+
+template<typename LevelSetType, typename FluidMechanicsType>
+void
+SurfaceTensionForceModel<LevelSetType, FluidMechanicsType>::build( std::string const& prefix, levelset_ptrtype const& ls, fluidmechanics_ptrtype const& fm )
+{
+    super_type::build( prefix, ls, fm );
+    this->loadParametersFromOptionsVm();
+}
+
+template<typename LevelSetType, typename FluidMechanicsType>
+std::shared_ptr<std::ostringstream> 
+SurfaceTensionForceModel<LevelSetType, FluidMechanicsType>::getInfo() const
+{
+    std::shared_ptr<std::ostringstream> _ostr( new std::ostringstream() );
+    *_ostr << "Surface tension ("
+           << "coeff = " << this->M_surfaceTensionCoeff
+           << ")";
+
+    return _ostr;
+}
+
+template<typename LevelSetType, typename FluidMechanicsType>
+void
+SurfaceTensionForceModel<LevelSetType, FluidMechanicsType>::loadParametersFromOptionsVm()
+{
+    M_surfaceTensionCoeff = doption( _name="surface-tension.coeff", _prefix=this->prefix() ); 
```

Comment:
do not use option in classes directly but set them with command line in construction. We should be able to modify this kind of option from the interface. with your code we can do that only from the command line

Reply:
this may be done elsewhere, I have checked thoroughly. This is a bad practice (that I started) which needs to be changed

Reply:
@prudhomm loadParametersFromOptionsVm() is method define by all toolboxes, it's call only by the constructuor and of course we can changes the value after